scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/TaskDBUtils.java

```diff
@@ -0,0 +1,150 @@
+import org.ow2.proactive.scheduler.core.db.TransactionHelper.SessionWork;
+
+
+public class TaskDBUtils {
+
+    public static SessionWork<List<TaskState>> taskStateSessionWork(final DBTaskDataParameters params) {
+
+        return new SessionWork<List<TaskState>>() {
+
+            @Override
+            @SuppressWarnings("unchecked")
+            public List<TaskState> executeWork(Session session) {
+                Criteria criteria = session.createCriteria(TaskData.class);
+
+                if (params.getLimit() > 0)
+                    criteria.setMaxResults(params.getLimit());
+                if (params.getOffset() >= 0)
+                    criteria.setFirstResult(params.getOffset());
+                if (params.getSortParameters() != null) {
+                    Order sortOrder;
+                    for (SortParameter<TaskSortParameter> param : params.getSortParameters()) {
+                        switch (param.getParameter()) {
+                            case ID:
+                                sortOrder = configureSortOrderTask(param, Property.forName("id"));
+                                break;
+                            case NAME:
+                                sortOrder = configureSortOrderTask(param, Property.forName("name"));
+                                break;
+                            default:
+                                throw new IllegalArgumentException(
+                                    "Unsupported sort parameter: " + param.getParameter());
+                        }
+                        criteria.addOrder(sortOrder);
+                    }
+                }
+
+                if (params.getUser() != null && "".compareTo(params.getUser()) != 0) {
+                    criteria.createAlias("jobData", "job")
+                            .add(Restrictions.eq("job.owner", params.getUser()));
+                }
+
+                if (params.getFrom() != 0 && params.getTo() != 0) {
+                    criteria.add(Restrictions.ge("startTime", params.getFrom()));
+                    criteria.add(Restrictions.le("finishedTime", params.getTo()));
+                }
+
+                if (!params.getStatuses().isEmpty()) {
+                    criteria.add(Restrictions.in("taskStatus", params.getStatuses()));
+                }
+
+                if (params.getTag() != null && "".compareTo(params.getTag()) != 0) {
+                    criteria.add(Restrictions.eq("tag", params.getTag()));
+                }
+
+                List<TaskData> tasksList = criteria.list();
+                List<TaskState> result = new ArrayList<TaskState>(tasksList.size());
+                for (TaskData taskData : tasksList) {
+                    result.add(taskData.toTaskState());
+                }
+                return result;
+            }
+        };
+
+    }
+    
+    public static SessionWork<List<TaskInfo>> taskInfoSessionWork(final DBTaskDataParameters params) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "taskInfoSessionWork" is 18 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)